### ISSUE TITLE
Fix "HPHP::jit::findMainExitBlock(const HPHP::jit::IRUnit&, HPHP::SrcKey): assertion ` mainExit' failed." on hphp/test/quick/surprise_throw.php

### ISSUE THREAD

**Issue by gut** [PPC64/hhvm#122]
Testing `hphp/test/quick/surprise_throw.php`. Currently it outputs:

```
hi exit fb_setprofile
hhvm: /root/hhvm/hphp/runtime/vm/bytecode.cpp:193: bool HPHP::isVMFrame(const HPHP::ActRec*): Assertion `ar' failed.
hhvm: /root/hhvm/hphp/runtime/vm/bytecode.cpp:193: bool HPHP::isVMFrame(const HPHP::ActRec*): Assertion `ar' failed.
Aborted
```

That happens on the _merge_official_01022016_ branch (2737381). Probably due to stub ABI updates

**Comment by gut** [comment]
Currently ( 69fb69fd705597a0f657cab003f9ae18f65cfdff ) the test fails like this:

```
hi exit fb_setprofile
hi enter foo

Assertion failure: /root/hhvm/hphp/runtime/vm/jit/ir-unit.cpp:170: HPHP::jit::Block* HPHP::jit::findMainExitBlock(const HPHP::jit::IRUnit&, HPHP::SrcKey): assertion `mainE
xit' failed.
findMainExit: no exit found for lastSk = foo(id 0x18c0)@186

-----------------------------------RegionDesc-----------------------------------
Region (1 blocks):
Block -10 foo@141 length 19 initSpOff 3 profTransID -1
  precondition: Local{0} :: Uninit (DataTypeCountness)
  precondition: Local{1} :: Uninit (DataTypeCountness)
  precondition: Local{2} :: Str (DataTypeCountness)
    141  FPushCtorD 0 "stdclass"
    147  FCall 0                                  (top func: stdClass::86ctor)
    149  PopR
    150  SetL L:0
    152  PopC
    153  FPushCtorD 0 "stdclass"
    159  FCall 0                                  (top func: stdClass::86ctor)
    161  PopR
    162  SetL L:1
    164  PopC
    165  FPushCtorD 0 "stdclass"
    171  FCall 0                                  (top func: stdClass::86ctor)
    173  PopR
    174  SetL L:2
    176  PopC
    177  FPushCtorD 0 "stdclass"
    183  FCall 0                                  (top func: stdClass::86ctor)
    185  PopR
    186  RetC

digraph RegionCFG {
 node[shape=box,style=filled]
 "B-10" [label="B -10\np: 0",fillcolor="#ffffff"]
}

-------------------------------------IRUnit-------------------------------------
digraph G {
B0 -> B1
B1 -> B4 [color=cyan]
}

 B0: [profCount=1]
Function (mayusevv) foo at 141
maxStackCells: 9
numLocals: 3
numIterators: 0
    --- bc 141, fp _, spOff 3 (foo)
 141: FPushCtorD 0 "stdclass"
    (00) t0:FramePtr = DefFP
    (01) t1:StkPtr = DefSP<FPRelOff 3> t0:FramePtr
    (02) Jmp -> B1
  no fallthrough

 B1: [profCount=1] (preds B0)
    --- bc 141, fp 0, spOff 3 (foo)
 141: FPushCtorD 0 "stdclass"
    (03) AssertLoc<Uninit,0> t0:FramePtr
    (04) AssertLoc<Uninit,1> t0:FramePtr
    (05) AssertLoc<Uninit,2> t0:FramePtr
    (09) Jmp -> B4<Unlikely>
  no fallthrough

------------------------cold blocks-------------------------
 B4<Unlikely>: [profCount=1]<Unlikely> (preds B1)
    (08) ReqRetranslate<0,0> t1:StkPtr, t0:FramePtr
  no fallthrough



Assertion failure: /root/hhvm/hphp/runtime/vm/jit/ir-unit.cpp:170: HPHP::jit::Block* HPHP::jit::findMainExitBlock(const HPHP::jit::IRUnit&, HPHP::SrcKey): assertion `mainExit' failed.
findMainExit: no exit found for lastSk = foo(id 0x18c0)@186
Segmentation fault
```

It should be simply:

```
hi exit fb_setprofile
hi enter foo
yep
hi enter Exception::getMessage
hi exit Exception::getMessage
yo
hi exit
```

**Comment by gut** [comment]
solved by 66442d8d439b64f441724b673ad51eef912f5109
